src/umpire/strategy/AlignedAllocator.cpp

```diff
@@ -0,0 +1,72 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/strategy/AlignedAllocator.hpp"
+
+#include "umpire/util/Macros.hpp"
+
+namespace umpire {
+namespace strategy {
+
+AlignedAllocator::AlignedAllocator(
+    const std::string& name,
+    int id,
+    Allocator allocator,
+    std::size_t alignment) :
+  AllocationStrategy(name, id),
+  m_allocator(allocator.getAllocationStrategy()),
+  m_alignment{alignment},
+  m_mask{static_cast<uintptr_t>(~(m_alignment-1))}
+{
+  if (m_allocator->getPlatform() != Platform::host) {
+    UMPIRE_ERROR("Cannot construct AlignedAllocator from non-host Allocator.");
+  }
+
+  if (! (m_alignment > 0 && ((m_alignment & (m_alignment-1)) == 0))) {
+    UMPIRE_ERROR("AlignedAllocator alignment must be a power of 2");
+  }
+}
+
+void* 
+AlignedAllocator::allocate(std::size_t bytes)
+{
+  std::size_t total_bytes = bytes+sizeof(void*)+m_alignment-1;
+  UMPIRE_LOG(Debug, "requested: " << bytes << " actual: " << bytes+m_alignment-1);
+
+  uintptr_t ptr{reinterpret_cast<uintptr_t>(m_allocator->allocate(total_bytes))};
+  uintptr_t aligned_ptr{static_cast<uintptr_t>((reinterpret_cast<uintptr_t>(ptr) + sizeof(void*) + (m_alignment-1)) & m_mask)}; 
+  uintptr_t* header = (uintptr_t*) (aligned_ptr - sizeof(void*));
```

Comment:
This line assumes that the minimum alignment size is whatever is required for a uintptr_t*.  I believe that this means that the alignment cannot be 1, 2, or (possibly) 4 (it certainly cannot be 1 :-)).  I suggest that we update the constructor to make certain the alignment is a power of 2 that is >= 8.